components/layout_2020/flow/mod.rs

```diff
@@ -183,15 +183,15 @@ impl BlockFormattingContext {
         // The content height of a BFC root should include any float participating in that BFC
         // (https://drafts.csswg.org/css2/#root-height), we implement this by imagining there is
         // an element with `clear: both` after the actual contents.
-        let clearance = sequential_layout_state.map_or(Length::zero(), |sequential_layout_state| {
+        let clearance = sequential_layout_state.and_then(|sequential_layout_state| {
             sequential_layout_state.calculate_clearance(ClearSide::Both)
         });
 
         IndependentLayout {
             fragments: flow_layout.fragments,
             content_block_size: flow_layout.content_block_size +
                 flow_layout.collapsible_margins_in_children.end.solve() +
-                clearance,
```

Comment:
It's slightly more efficient to do `clearance.unwrap_or_else(Length::zero)` here.